### ISSUE TITLE
docker-compose up error: /usr/bin/prepare.sh: No such file or directory 

### ISSUE THREAD

**Issue by f-guitart** [dask/dask-docker#38]
I am trying to compose up the three docker images but I get the following errors:

```
C:\cygwin64\home\fcgr\code\dask-docker>docker-compose up
Starting dask-docker_worker_1     ... done
Starting dask-docker_scheduler_1  ... done
Recreating dask-docker_notebook_1 ... done
Attaching to dask-docker_worker_1, dask-docker_notebook_1, dask-docker_scheduler_1
worker_1       |  [dumb-init] /usr/bin/prepare.sh: No such file or directory
notebook_1   |  [FATAL tini (6)] exec /usr/bin/prepare.sh failed: No such file or directory
scheduler_1   |  [dumb-init] /usr/bin/prepare.sh: No such file or directory
dask-docker_worker_1 exited with code 2
dask-docker_notebook_1 exited with code 127
dask-docker_scheduler_1 exited with code 2
```

I am using Docker Version 18.06.1-ce-win73

**Comment by f-guitart** [comment]
After digging a little bit I found that Windows style newline characters can cause issues when copying prepare.sh to the docker container. I don't understand when these newline characters are introduced. I guess that at some point '\n' are converted to '\r'

What I did was to directly download prepare.sh from github using wget, and then copying to /usr/bin. This works for me.

I don't know if there is any effective way to assure that no '\r' is introduced.

**Comment by mrocklin** [comment]
Thank you for raising this.  I also don't know how to address this.

**Comment by asherp** [comment]
Hey, I'm also on windows and found this error. I thought it was the newline character too, but switching to unix line-endings didn't fix it.

However, I found some other post where the issue was related to dumb-init getting confused by the bash script. The solution is to break up  the entrypoint and command lines:

dask-docker/base/Dockerfile
```
ENTRYPOINT ["/usr/local/bin/dumb-init"]
CMD ["/usr/bin/prepare.sh"]
```
dask-docker/notebook/Dockerfile
```
ENTRYPOINT ["tini", "--"]
CMD ["/usr/bin/prepare.sh"]
CMD ["start.sh", "jupyter", "lab"]
```

**Comment by detroyejr** [comment]
I just ran into this issue today and fixed it by changing the line endings. It looks like git allows you to [force which line endings certain files need to have through a .gitattributes file.](https://help.github.com/en/github/using-git/configuring-git-to-handle-line-endings) I tested this and `git clone` checked out base/prepare.sh with the proper LF style line endings. After that, building and running the container worked great.

**Comment by jrbourbeau** [comment]
Thanks for looking into this @detroyejr! Would this be something we could add to `dask-docker`, or should is this something users should handle locally?

**Comment by detroyejr** [comment]
Submitted a PR. Assuming that the entrypoint scripts are the only things causing trouble this should fix the problem. If there are other line-ending issues, then we can add more file types.

**Comment by TomAugspurger** [comment]
Closed by #63
